### PR TITLE
Merge sequential checks

### PR DESCRIPTION
**Pull request by gpetrou**

### CODE REVIEW

File: Src/Microsoft.Dynamic/zlib.net/Inflate.cs [hunk 1]
```diff
@@ -259,12 +259,12 @@ internal int inflateInit(ZStream z, int windowBits)
 		internal int inflate(ZStream z, FlushStrategy flush)
 		{
 		    int internalFlush = (int)flush;
-
-            int res_temp;
 			
-			if (z == null || z.istate == null || z.next_in == null)
+			if (z?.istate == null || z.next_in == null)
```
Comment (slozier) [inline Src/Microsoft.Dynamic/zlib.net/Inflate.cs]: I'd prefer keeping the null check separate as it's clearer that it's also not null in the second condition.

File: Src/Microsoft.Dynamic/zlib.net/Inflate.cs [hunk 2]
```diff
@@ -487,7 +487,7 @@ internal int inflateSetDictionary(ZStream z, byte[] dictionary, int dictLength)
 		{
 			int index = 0;
 			int length = dictLength;
-			if (z == null || z.istate == null || z.istate.mode !=  InflateMode.DICT0)
+			if (z?.istate == null || z.istate.mode !=  InflateMode.DICT0)
```
Comment (slozier) [inline Src/Microsoft.Dynamic/zlib.net/Inflate.cs]: Does `z?.istate?.mode != ...` work?

**Comment by gpetrou** [comment]
I also replaced some tabs with spaces. Seems like some of these files have mixed tabs with spaces.

File: Src/Microsoft.Dynamic/zlib.net/Inflate.cs [hunk 3]
```diff
@@ -487,7 +487,7 @@ internal int inflateSetDictionary(ZStream z, byte[] dictionary, int dictLength)
 		{
 			int index = 0;
 			int length = dictLength;
-			if (z == null || z.istate == null || z.istate.mode !=  InflateMode.DICT0)
+			if (z?.istate?.mode != InflateMode.DICT0)
```
Comment (slide) [inline Src/Microsoft.Dynamic/zlib.net/Inflate.cs]: Doesn't this change the meaning of the if check?

Comment (slozier) [inline Src/Microsoft.Dynamic/zlib.net/Inflate.cs]: If z or z.istate are null then z?.istate?.mode is null so it's != InflateMode.DICT0. I agree it's not the most obvious conversion.

Comment (slide) [inline Src/Microsoft.Dynamic/zlib.net/Inflate.cs]: All these changes make me want to go back in my SW at work and actually use new features of C# :-)

**Pull request merged**